File: Summation of four primes/sumOf4Prime.cpp
#include <bits/stdc++.h> 
using namespace std; 

int isPrime(int x)  { 
	int s = sqrt(x); 

	for (int i = 2; i <= s; i++) {
		if (x % i == 0) 
			return 0; 
	}
	return 1; 
} 

void Num(int x, int &a, int &b) { 
	for (int i = 2; i <= x / 2; i++) { 
		if (isPrime(i) && isPrime(x - i)) { 
			a = i; 
			b = x - i; 
			return; 
		} 
	} 
} 

void generate(int n) { 
	if (n <= 7) 
		cout << "Impossible to form" << endl; 

	int a, b; 

	if (n % 2 != 0) { 
		Num(n - 5, a, b); 
		cout << "2 3 " << a << " " << b << endl; 
	} 

	else { 

		Num(n - 4, a, b); 

		cout << "2 2 " << a << " " << b << endl; 
	} 
} 

int main() { 
	int n;
	cout << "Enter n: ";
	cin >> n;
	generate(n); 
	return 0; 
} 

File: Summation of four primes/sumOf4Prime.py
import math

def is_prime(x):
    s = int(math.sqrt(x))
    for i in range(2, s + 1):
        if x % i == 0:
            return False
    return True

def Num(x):
    for i in range(2, x // 2 + 1):
        if is_prime(i) and is_prime(x - i):
            return [i, x - i]

def generate(n):
    if n <= 7:
        print("Impossible to form")
        return

    ab = [0, 0]

    if n % 2 != 0:
        ab = Num(n - 5)
        print("2 3 {} {}".format(ab[0], ab[1]))
    else:
        ab = Num(n - 4)
        print("2 2 {} {}".format(ab[0], ab[1]))

if __name__ == "__main__":
    n = int(input("Enter n: "))
    generate(n)
